expression/builtin_ilike_vec.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package expression
+
+import (
+	"github.com/pingcap/tidb/util/chunk"
+	"github.com/pingcap/tidb/util/collate"
+	"github.com/pingcap/tidb/util/hack"
+	"github.com/pingcap/tidb/util/stringutil"
+)
+
+func LowerAlphaAscii(lowered_col *chunk.Column, row_num int) {
+	for i := 0; i < row_num; i++ {
+		str := lowered_col.GetString(i)
+		str_bytes := hack.Slice(str)
+
+		stringutil.LowerOneString(str_bytes)
+	}
+}
+
+func LowerAlphaAsciiExcludeEscapeChar(lowered_col *chunk.Column, row_num int, excluded_char []int64) {
+	for i := 0; i < row_num; i++ {
+		str := lowered_col.GetString(i)
+		str_bytes := hack.Slice(str)
+		escape_char := excluded_char[i]
+
+		excluded_char[i] = int64(stringutil.LowerOneStringExcludeEscapeChar(str_bytes, byte(escape_char)))
+	}
+}
+
+func (b *builtinIlikeSig) vectorized() bool {
+	return true
+}
+
+func (b *builtinIlikeSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
+	rowNum := input.NumRows()
+	bufVal, err := b.bufAllocator.get()
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(bufVal)
+	if err = b.args[0].VecEvalString(b.ctx, input, bufVal); err != nil {
+		return err
+	}
+	bufPattern, err := b.bufAllocator.get()
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(bufPattern)
+	if err = b.args[1].VecEvalString(b.ctx, input, bufPattern); err != nil {
+		return err
+	}
+
+	bufEscape, err := b.bufAllocator.get()
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(bufEscape)
+	if err = b.args[2].VecEvalInt(b.ctx, input, bufEscape); err != nil {
+		return err
+	}
+	escapes := bufEscape.Int64s()
+
+	// Must not use b.pattern to avoid data race
+	pattern := collate.ConvertAndGetBinCollation(b.collation).Pattern()
+
+	tmpValCol := bufVal.CopyConstruct(nil)
+	tmpPatternCol := bufPattern.CopyConstruct(nil)
+	LowerAlphaAscii(tmpValCol, rowNum)
+	LowerAlphaAsciiExcludeEscapeChar(tmpPatternCol, rowNum, escapes)
```

Comment:
will `escape` column be modified inplace?

Reply:
> will `escape` column be modified inplace?

yes, and i will fix it